artiq/firmware/libproto_artiq/drtioaux_proto.rs

```diff
@@ -57,6 +59,15 @@ pub enum Packet {
 
     JdacBasicRequest { destination: u8, dacno: u8, reqno: u8, param: u8 },
     JdacBasicReply { succeeded: bool, retval: u8 },
+
+    //maximum DMA_TRACE_MAX_SIZE to fit within 512 max size + leeway
+    DmaAddTraceRequest { id: u32, last: bool, length: u32, trace: [u8; DMA_TRACE_MAX_SIZE] },
```

Comment:
I wanted to use ``Vec<u8>`` here - however it does not implement ``Copy`` trait and as is is not compatible with this :X So instead, there's DMA_TRACE_MAX_SIZE array preallocated, but only necessary bytes are sent and received.